authentication/src/main/java/io/camunda/authentication/config/WebSecurityConfig.java

```diff
@@ -129,50 +128,117 @@ private void genericSuccessHandler(
-          .authorizeHttpRequests(
-              (authorizeHttpRequests) ->
-                  authorizeHttpRequests
-                      .requestMatchers(UNAUTHENTICATED_PATHS)
-                      .permitAll()
-                      .anyRequest()
-                      .authenticated())
-          .headers(
-              (headers) ->
-                  headers.httpStrictTransportSecurity(
-                      (httpStrictTransportSecurity) ->
-                          httpStrictTransportSecurity
-                              .includeSubDomains(true)
-                              .maxAgeInSeconds(63072000)
-                              .preload(true)))
-          .exceptionHandling(
-              (exceptionHandling) -> exceptionHandling.accessDeniedHandler(authFailureHandler))
-          .csrf(AbstractHttpConfigurer::disable)
-          .cors(AbstractHttpConfigurer::disable)
-          .formLogin(AbstractHttpConfigurer::disable)
-          .anonymous(AbstractHttpConfigurer::disable);
-    } catch (final Exception e) {
-      throw new RuntimeException(e);
-    }
+  @Bean
+  @Profile("auth-basic")
+  @Order(2)
+  public SecurityFilterChain loginAuthSecurityFilterChain(
+      final HttpSecurity httpSecurity, final AuthFailureHandler authFailureHandler)
+      throws Exception {
+    return buildLoginAuthSecurityFilterChain(httpSecurity, authFailureHandler);
+  }
+
+  public SecurityFilterChain buildLoginAuthSecurityFilterChain(
+      final HttpSecurity httpSecurity, final AuthFailureHandler authFailureHandler)
+      throws Exception {
+    LOG.info("Configuring login auth");
+    return baseHttpSecurity(httpSecurity, authFailureHandler)
+        .formLogin(
+            formLogin ->
+                formLogin
+                    .loginPage(LOGIN_URL)
+                    .loginProcessingUrl(LOGIN_URL)
+                    .failureHandler(authFailureHandler)
+                    .successHandler(this::genericSuccessHandler))
+        .logout(
+            (logout) ->
+                logout
+                    .logoutUrl(LOGOUT_URL)
+                    .logoutSuccessHandler(this::genericSuccessHandler)
+                    .deleteCookies(SESSION_COOKIE))
+        .exceptionHandling(
+            exceptionHandling ->
+                exceptionHandling
+                    .authenticationEntryPoint(authFailureHandler)
+                    .accessDeniedHandler(authFailureHandler))
+        .build();
   }
 
   @Bean
   public FilterRegistrationBean<TenantRequestAttributeFilter>
       tenantRequestAttributeFilterRegistration(final MultiTenancyConfiguration configuration) {
     return new FilterRegistrationBean<>(new TenantRequestAttributeFilter(configuration));
   }
+
+  private static boolean isBasicAuthRequest(final HttpServletRequest request) {
+    final String authorizationHeader = request.getHeader(HttpHeaders.AUTHORIZATION);
+    // If the client sends an `Authorization: Basic` header, treat this as a basic auth request.
```

Comment:
💭 NABD but I typically try to avoid comments unless its something difficult to understand or has a specific not visible purpose. I think some of these comments don't meet that criteria for me but I'm not going to push on this :)

Reply:
I've added this comment so the whole purpose of the method is explained (including this boring base-case). In general, I find there are too few comments in this code base, and maybe that makes these feel out of place. Would it be better as a Javadoc comment?

Reply:
I don't think a javadoc would be the right option here (typically javadocs are only used for the public code here i.e. the java client.

Its fine to leave them in to me but yea maybe its because of fewer comments elsewhere that make it a flag for me :D